src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -1585,7 +1597,134 @@ else if (ticket == TicketType.POST_TELEPORT) // post teleport
                 cc.s.getLevel().getChunkSource().addRegionTicket(TicketType.UNKNOWN, target, radius, target);
             return new NumericValue(ticket.timeout());
         });
+        // =======================================
+        expression.addContextFunction("save_structure_template", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            ServerLevel world = cc.s.getLevel();
+
+            StructureManager structureManager = world.getStructureManager();
+            StructureTemplate theStructure;
+            String name;
+
+            boolean returnnbt = lv.get(0).isNull();
+            ResourceLocation struident = null;
+            if (!returnnbt) {
+                name = lv.get(0).getString();
+                struident = ResourceLocation.tryParse(name);
+                if (name.isEmpty() || name.endsWith(":") || struident == null) {
+                    return Value.NULL;
+                } // if(lv.size()==2&&lv.get(1).isNull()){lv3.remove(struident);return
+                  // Value.NULL;}
+                  theStructure = structureManager.getOrCreate(struident);
+            } else {
+                theStructure = new StructureTemplate();
+            }
+            BlockArgument start = BlockArgument.findIn((CarpetContext) c, lv, 1);
+            BlockArgument end = BlockArgument.findIn((CarpetContext) c, lv, start.offset);
+            BlockArgument ignoredblock = BlockArgument.findIn((CarpetContext) c, lv, end.offset + 1, true, true, false);
```

Comment:
ignored block should for sure be optional - most usecases wouldn't require it, so please count your lv's and only read it when needed.

Reply:
they are all optional now.